pom.xml

```diff
@@ -1748,7 +1684,113 @@ flexible messaging model and an intuitive client API.</description>
       <properties>
         <rename.netty.native.libs>rename-netty-native-libs.cmd</rename.netty.native.libs>
       </properties>
+
+    <!-- Primary Module profile -->
+    </profile>
+      <profile>
+      <id>main</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <modules>
+        <module>buildtools</module>
+        <module>testmocks</module>
+        <module>managed-ledger</module>
+        <module>tiered-storage</module>
+        <module>pulsar-common</module>
+        <module>pulsar-broker-common</module>
+        <module>pulsar-broker</module>
+        <module>pulsar-broker-shaded</module>
+        <module>pulsar-client-api</module>
+        <module>pulsar-client</module>
+        <module>pulsar-client-shaded</module>
+        <module>pulsar-client-1x-base</module>
+        <module>pulsar-client-admin</module>
+        <module>pulsar-client-admin-shaded</module>
+        <module>pulsar-client-tools</module>
+        <module>pulsar-client-tools-test</module>
+        <module>pulsar-client-all</module>
+        <module>pulsar-websocket</module>
+        <module>pulsar-proxy</module>
+        <module>pulsar-discovery-service</module>
+        <module>pulsar-storm</module>
+        <module>pulsar-flink</module>
+        <module>pulsar-spark</module>
+        <module>pulsar-zookeeper-utils</module>
+        <module>pulsar-testclient</module>
+        <module>pulsar-broker-auth-athenz</module>
+        <module>pulsar-client-auth-athenz</module>
+        <module>pulsar-client-kafka-compat</module>
+        <module>pulsar-zookeeper</module>
+        <module>pulsar-log4j2-appender</module>
+        <module>pulsar-sql</module>
+        <module>dashboard</module>
+        <module>pulsar-broker-auth-sasl</module>
+        <module>pulsar-client-auth-sasl</module>
+
+        <!-- transaction related modules -->
+        <module>pulsar-transaction</module>
+
+        <!-- functions-related modules -->
+        <module>pulsar-functions</module>
+
+        <!-- connector-related modules -->
+        <module>pulsar-io</module>
+
+        <!-- kafka connect avro converter shaded, because version mismatch    -->
+        <module>kafka-connect-avro-converter-shaded</module>
+
+        <!-- examples -->
+        <module>examples</module>
+
+        <!-- Bouncy Castle Provider loaders-->
+        <module>bouncy-castle</module>
+
+        <module>pulsar-client-messagecrypto-bc</module>
+
+        <!-- all these 3 modules should be put at the end in this exact sequence -->
+        <module>distribution</module>
+        <module>docker</module>
+        <module>tests</module>
+        <module>pulsar-metadata</module>
+        <module>jclouds-shaded</module>
+      </modules>
     </profile>
+
+    <!-- core profile focused of pulsar java modules -->
+    <profile>
+      <id>core-modules</id>
+      <modules>
+        <module>buildtools</module>
```

Comment:
Ideally, it would be better to not have to repeat the list of `core-modules` in both lists. I'm not sure that is possible.

In absence of that, we could just make it visually clear, with the list of core-modules sorted in same way and with spaces (and comments) from the extra modules.